src/engine/qml.js

```diff
@@ -234,15 +234,21 @@ function construct(meta) {
         // Currently we support only 1,2 and 4 and use order: 4,1,2
         // TODO: engine.qmldirs is global for all loaded components. That's not qml's original behaviour.
         var qdirInfo = engine.qmldirs[meta.object.$class]; // Are we have info on that component in some imported qmldir files?
+
+        var oldExecutionContext = _executionContext;
+        _executionContext = meta.context;
+
         if (qdirInfo) {
             // We have that component in some qmldir, load it from qmldir's url
             component = Qt.createComponent( "@" + qdirInfo.url);
         }
         else
             component = Qt.createComponent(meta.object.$class + ".qml");
 
+        _executionContext = oldExecutionContext;
```

Comment:
@akreuzkamp, yes that should work fine. Though I think we need to keep line 365, as `applyProperties` is called manually by the QMLListElement constructor.